### ISSUE TITLE
HTTP parser/serializer error

### ISSUE THREAD

**Issue by Joannis** [vapor/engine#218]
```swift
router.get("json") { req -> Future<Response> in
	let client = try req.make(Client.self)
	
	// Return the response from this URL
	return client.get("http://example.vapor.codes/json")
}
```

The above should be returning the HTTP response from example.vapor.codes to Vapor's client.

Instead a response with the following body appears:

```
Content-Length: 87

{"array":[0,1,2,3],"dict":{"lang":"Swift","name":"Vapor"},"number":123,"string":"test"}
```

This content-length header should be part of the headers section but is found as part of the body section instead.

**Comment by twof** [comment]
Related?
https://github.com/vapor/vapor/issues/1481

**Comment by tanner0101** [comment]
I believe this was fixed with #220 (I remember seeing something about an extraneous /r/n which would explain this)
